### ISSUE TITLE
Create a jar

### ISSUE THREAD

**Issue by adhamarif** [till-teb/expenses-management-tool#21]
Other than expense tracker, maybe we could create a jar to track our savings. This will be useful to compare our expense and savings every month.

Features:

1. Integrate with expense categories `savings`
2. Set a goal for savings (example 1000$ in a year), then it shows the remaining time and balance.
3. Shows prediction

Additional features:
This is not a must but I find this feature really cool.

- A financial suggestion that correlate with expenses tracker. Let's say it's called **Guilty Pleasure**
- It has 2 main component, which is `spend_limit` and `compensation`
- Now the user set the `spend_limit` to spend only 100$ in `Entertainment` and 10$ for `compensation` . If they exceed the limit, then a prompt/notifications will appear that user should compensate 10$ in every transaction they made in `Entertainment` to their savings.

What do you guys think about my idea? Im looking forward to your comment and suggestion!!

**Comment by l4nz8** [comment]
I think this is a good idea, to implement this we could either, each time the program is started, all previous spending history, match with our income history and output as our savings. But that would be quite performance heavy because at startup the whole history would have to be converted to give us that. Another way to output savings would be to calculate them for each month and save them in a table at the end of the month. This would also be good to use at the end when you have more exceptions than you actually have in the month and would be considered as a kind of credit.

**Comment by adhamarif** [comment]
Yup I agreed on that. We should be focusing more on the development of this program first, and then we will be working on this as extra feature if we have enough time.

Anyway thank you for your suggestion! 😁
